bentoctl/docker_utils.py

```diff
@@ -94,49 +72,27 @@ def generate_deployable_container(
         buildx.build(**buildx_args)
 
 
-def tag_docker_image(image_name, image_tag):
-    docker_client = docker.from_env()
-    try:
-        img = docker_client.images.get(image_name)
-        was_tagged = img.tag(image_tag)
-        if not was_tagged:
-            raise BentoctlDockerException(
-                "Failed to tag docker image! tag function returned False"
-            )
-    except docker.errors.ImageNotFound:
-        raise BentoctlDockerException(
-            f"Failed to tag Docker image, {image_name} not found."
-        )
-    except docker.errors.APIError as error:
-        raise BentoctlDockerException(
-            f"Failed to tag docker image {image_tag}: {error}"
-        )
+def generate_tag(image_name: str, image_tag: str) -> None:
+    run_subprocess_cmd(["docker", "tag", image_name, image_tag])
 
 
-def push_docker_image_to_repository(
-    repository, image_tag=None, username=None, password=None
-):
-    docker_client = docker.from_env()
-    docker_push_kwags = {"repository": repository, "tag": image_tag}
-    if username is not None and password is not None:
-        docker_push_kwags["auth_config"] = {"username": username, "password": password}
-    try:
-        progress_bar = DockerPushProgressBar()
-        with Live(progress_bar) as live:
-            for line in docker_client.images.push(
-                **docker_push_kwags, decode=True, stream=True
-            ):
-                if "id" in line:
-                    progress_bar.update(line)
-                    live.update(progress_bar)
-                elif "status" in line:
-                    print(line.get("status"))
-                elif "errorDetail" in line:
-                    raise BentoctlDockerException(
-                        f"Failed to push docker image. {line['error']}"
-                    )
-        console.print(":rocket: Image pushed!")
-    except docker.errors.APIError as error:
-        raise BentoctlDockerException(
-            f"Failed to push docker image {image_tag}: {error}"
-        )
+def push_to_repository(
+    image_tag: str,
+    disable_content_trust: bool | None = None,
+    *,
+    username: str | None = None,
+    password: str | None = None,
+) -> None:
+    if username:
+        assert password, "password is required when username is provided"
+        login_cmd = ["docker", "login"]
+        login_cmd.extend(["-u", username, "-p", password])
+        run_subprocess_cmd(login_cmd)
```

Comment:
Is there a way we pass in username and password only when we are pushing the image?

I don't want to override user's current logged in account on docker CLI